src/modules/DefaultDesktop.rb

```diff
@@ -25,153 +26,47 @@ def main
       @desktop = nil
 
       @initialized = false
-
-      @packages_proposal_ID_patterns = "DefaultDesktopPatterns"
     end
 
-    def MissingKey(desktop_def, key)
-      Builtins.y2warning(
-        "Wrong desktop def, missing '%1' key: %2",
-        key,
-        desktop_def.value
-      )
+    # Defaults when something in control file is not defined
+    DEFAULT_VALUES = {
+      "order"   => 99,
+      "desktop" => "unknown"
+    }
 
-      case key
-      when "order"
-        99
-      when "desktop"
-        "unknown"
-      else
-        ""
-      end
-    end
+    DEFAULT_VALUES.default_proc = Proc.new { "" }
```

Comment:
NP: The `default_proc` is not common in the YaST code, I'd suggest add a comment like
```Ruby
# define the default value for missing keys
```